homeassistant/components/adax/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for Adax integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import adax
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import ACCOUNT_ID, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {vol.Required(ACCOUNT_ID): int, vol.Required(CONF_PASSWORD): str}
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect."""
+    account_id = data[ACCOUNT_ID]
+    password = data[CONF_PASSWORD].replace(" ", "")
+
+    token = await adax.get_adax_token(
+        async_get_clientsession(hass), account_id, password
+    )
+    if token is None:
+        _LOGGER.info("Adax: Failed to login to retrieve token")
+        raise CannotConnect
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Adax."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            self._async_abort_entries_match({ACCOUNT_ID: user_input[ACCOUNT_ID]})
```

Comment:
Please only wrap the line that can raise in the try... except block.

Reply:
Maybe we can use the account id as unique id for the config entry?

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-id-requirements